# EEP/IAS 118

## Confidence Intervals for Prediction Example

### July 30, 2019


In [1]:
library(tidyverse)
library(haven)
library(stargazer)

sleepdata <- read_dta("sleep75.dta") %>%
    mutate(sleep_nightly = sleep/7/60)
head(sleepdata)


reg <- lm(sleep_nightly ~ educ + exper + hrwage , data = sleepdata)
summary(reg)
#stargazer(reg, type = "html")


Registered S3 methods overwritten by 'ggplot2':
  method         from 
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ ggplot2 3.1.0     ✔ purrr   0.3.2
✔ tibble  2.1.3     ✔ dplyr   0.8.1
✔ tidyr   0.8.3     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.4.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

Please cite as: 

 Hlavac, Marek (2018). stargazer: Well-Formatted Regression and Summary Statistics Tables.
 R package version 5.2.2. https://CRAN.R-project.org/package=stargazer 



age black case clerical construc educ earns74 gdhlth inlf leis1 ⋯ totwrk
1 32  0     1    0        0        12       0   0      1    3529  ⋯ 3438  
2 31  0     2    0        0        14    9500   1      1    2140  ⋯ 5020  
3 44  0     3    0        0        17   42500   1      1    4595  ⋯ 2815  
4 30  0     4    0        0        12   42500   1      1    3211  ⋯ 3786  
5 64  0     5    0        0        14    2500   1      1    4052  ⋯ 2580  
6 41  0     6    0        0        12       0   1      1    4812  ⋯ 1205  
  union worknrm workscnd exper yngkid yrsmarr hrwage    agesq sleep_nightly
1 0     3438       0     14    0      13       7.070004 1024  7.411905     
2 0     5020       0     11    0       0       1.429999  961  6.952381     
3 0     2815       0     21    0       0      20.529997 1936  6.357143     
4 0     3786       0     12    0      12       9.619998  900  7.340476     
5 0     2580       0     44    0      33       2.750000 4096  8.209524     
6 0        0    1205     23    0      23      19.249998 1681  9.673810


Call:
lm(formula = sleep_nightly ~ educ + exper + hrwage, data = sleepdata)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1833 -0.6140 -0.0052  0.5941  3.3239 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.880271   0.285872  27.566   <2e-16 ***
educ        -0.011255   0.019477  -0.578    0.564    
exper        0.004340   0.004157   1.044    0.297    
hrwage      -0.012069   0.012646  -0.954    0.340    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.025 on 528 degrees of freedom
  (174 observations deleted due to missingness)
Multiple R-squared:  0.007316,	Adjusted R-squared:  0.001675 
F-statistic: 1.297 on 3 and 528 DF,  p-value: 0.2747


<table style="text-align:center"><tr><td colspan="2" style="border-bottom: 1px solid black"></td></tr><tr><td style="text-align:left"></td><td><em>Dependent variable:</em></td></tr>
<tr><td></td><td colspan="1" style="border-bottom: 1px solid black"></td></tr>
<tr><td style="text-align:left"></td><td>sleep_nightly</td></tr>
<tr><td colspan="2" style="border-bottom: 1px solid black"></td></tr><tr><td style="text-align:left">educ</td><td>-0.011</td></tr>
<tr><td style="text-align:left"></td><td>(0.019)</td></tr>
<tr><td style="text-align:left"></td><td></td></tr>
<tr><td style="text-align:left">exper</td><td>0.004</td></tr>
<tr><td style="text-align:left"></td><td>(0.004)</td></tr>
<tr><td style="text-align:left"></td><td></td></tr>
<tr><td style="text-align:left">hrwage</td><td>-0.012</td></tr>
<tr><td style="text-align:left"></td><td>(0.013)</td></tr>
<tr><td style="text-align:left"></td><td></td></tr>
<tr><td style="text-align:left">Constant</td><td>7.880<sup>***</sup></td></tr>
<tr><td style="text-align:left"></td><td>(0.286)</td></tr>
<tr><td style="text-align:left"></td><td></td></tr>
<tr><td colspan="2" style="border-bottom: 1px solid black"></td></tr><tr><td style="text-align:left">Observations</td><td>532</td></tr>
<tr><td style="text-align:left">R<sup>2</sup></td><td>0.007</td></tr>
<tr><td style="text-align:left">Adjusted R<sup>2</sup></td><td>0.002</td></tr>
<tr><td style="text-align:left">Residual Std. Error</td><td>1.025 (df = 528)</td></tr>
<tr><td style="text-align:left">F Statistic</td><td>1.297 (df = 3; 528)</td></tr>
<tr><td colspan="2" style="border-bottom: 1px solid black"></td></tr><tr><td style="text-align:left"><em>Note:</em></td><td style="text-align:right"><sup>*</sup>p<0.1; <sup>**</sup>p<0.05; <sup>***</sup>p<0.01</td></tr>
</table>

__Residual Std. Error__ reports $\frac{\sum \hat u^2_i}{n-k-1} = \frac{SSR}{n-k-1}$. To see this:

In [2]:
resid <- reg$residuals
ssr <- sum(resid^2)
paste0("SSR = ", ssr)
var_u <- ssr/(532 - 3 - 1)
se_u <- sqrt(var_u)
paste0("sigma_hat = ", round(se_u,3))


[1] "SSR = 555.017963171349"

[1] "sigma_hat = 1.025"

#### Predict Average hours slept per night when $educ = 10$, $exper = 4$, and $hrwage = 15$

In [3]:
yhat <- reg$coefficients[1] + reg$coefficients[2]*10 + reg$coefficients[3]*4 + reg$coefficients[4]*15
paste0("Predicted Average Y when educ is 10, exper is 4, and hourly wage is 15 equals ", yhat)

sleepdata <- sleepdata %>%
    mutate(educ_adj = educ - 10,
          exper_adj = exper - 4,
          wage_adj = hrwage - 15)

reg_adj <- lm(sleep_nightly ~ educ_adj +exper_adj +wage_adj, data = sleepdata)
summary(reg_adj)

[1] "Predicted Average Y when educ is 10, exper is 4, and hourly wage is 15 equals 7.60403764743343"


Call:
lm(formula = sleep_nightly ~ educ_adj + exper_adj + wage_adj, 
    data = sleepdata)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1833 -0.6140 -0.0052  0.5941  3.3239 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.604038   0.187291  40.600   <2e-16 ***
educ_adj    -0.011255   0.019477  -0.578    0.564    
exper_adj    0.004340   0.004157   1.044    0.297    
wage_adj    -0.012069   0.012646  -0.954    0.340    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.025 on 528 degrees of freedom
  (174 observations deleted due to missingness)
Multiple R-squared:  0.007316,	Adjusted R-squared:  0.001675 
F-statistic: 1.297 on 3 and 528 DF,  p-value: 0.2747


You can grab the value of the residual standard error from your `lm` object using the `sigma()` function.

In [4]:
sigma(reg_adj)

[1] 1.025266